### PR TITLE
[ide-service] move ide config type define to ide-service-api

### PR DESCRIPTION
**Pull request by iQQBot**
## Description

This PR move `ide-config` type define to `ide-service-api` and changes `installer` to use this type

Also introduce new configmap `ide-config` (it's same with `server-ide-config`) in the future, we will remove `server-ide-config`

## Related Issue(s)

Fixes #

## How to test

1. it should be able open a new workspace in this preview environment. (it won't break `server`)
2. Follow https://github.com/gitpod-io/gitpod/pull/12815 this PR `how to test` section, check it doesn't break `ide-service`

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Review by akosyakov — APPROVED** [review]
code-wise looks good, I have not tried it yet

/hold

please unhold if somebody else tries it before and it works

File: components/ide-service/cmd/test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ var testCommand = &cobra.Command{
 	Version: Version,
 	Run: func(cmd *cobra.Command, args []string) {
 		cfg := getConfig()
-		conn, err := grpc.Dial(fmt.Sprintf("localhost%s", cfg.Server.Services.GRPC.Address), grpc.WithTransportCredentials(insecure.NewCredentials()))
```
Comment (mustard-mh) [inline components/ide-service/cmd/test.go:28]: 🙏 Thanks for fix this up

**Comment by mustard-mh** [comment]
Build all failed, will delete and try again tomorrow

**Comment by mustard-mh** [comment]
Component `ide-service` started and worked, but `kubectl exec` failed (not this PR's problem), so I use port-forward here

image

Diff from ide options from `production` (a) and `preview env` (b) looks good (latestImage in preview env use `WORKSPACE.yaml` 's value)

image

**Review by mustard-mh — APPROVED** [review]
👍

**Review by easyCZ — APPROVED** [review]
LGTM, thanks for decoupling it from `server`

**Pull request merged**